comictaggerlib/taggerwindow.py

```diff
@@ -215,15 +216,29 @@ def __init__(
 
         if config[0].Runtime_Options__type and isinstance(config[0].Runtime_Options__type[0], str):
             # respect the command line option tag type
-            config[0].internal__save_data_style = config[0].Runtime_Options__type[0]
+            config[0].internal__save_data_style = config[0].Runtime_Options__type
             config[0].internal__load_data_style = config[0].Runtime_Options__type[0]
 
-        if config[0].internal__save_data_style not in metadata_styles:
-            config[0].internal__save_data_style = list(metadata_styles.keys())[0]
+        for style in config[0].internal__save_data_style:
+            if style not in metadata_styles:
+                config[0].internal__save_data_style.remove(style)
         if config[0].internal__load_data_style not in metadata_styles:
             config[0].internal__load_data_style = list(metadata_styles.keys())[0]
-        self.save_data_style = config[0].internal__save_data_style
-        self.load_data_style = config[0].internal__load_data_style
+        self.save_data_styles: list[str] = config[0].internal__save_data_style
+        self.load_data_style: str = config[0].internal__load_data_style
+
+        # Add multiselect combobox
+        self.cbSaveDataStyle = CheckableComboBox()
+        self.cbSaveDataStyle.setToolTip("At least 1 save style is required")
+        # Add normal combobox for read style (TODO support multiple read styles)
+        self.cbLoadDataStyle = QtWidgets.QComboBox()
+
+        # Need to set minimum or source_style_formLayout will resize larger than 230px which will affect the
+        # file info box and cover image width underneath
+        self.cbLoadDataStyle.setMinimumWidth(100)
+        self.cbSaveDataStyle.setMinimumWidth(100)
+        self.source_style_formLayout.addRow("Read Style", self.cbLoadDataStyle)
+        self.source_style_formLayout.addRow("Modify Styles", self.cbSaveDataStyle)
```

Comment:
We can actually put all of this into the ui file.
Right click on the widget and click on promote
<img width="606" alt="Screenshot 2024-02-10 at 1 22 52 PM" src="https://github.com/comictagger/comictagger/assets/7547075/6dbbf20f-aaef-4f6c-9468-6b0d26ae16cd">

Reply:
I thought it would require a C header. Should have looked into it further...